app/engine/accountWatcher.ts

```diff
@@ -33,14 +43,26 @@ export function useBlocksWatcher() {
             lastBlockResolve?.(data.seqno);
             lastBlockResolve = undefined;
             lastBlockPromise = Promise.resolve(data.seqno);
+
+            const lastBlockKey = `lastBlock-${selectedAccount?.addressString || ''}`;
+            const lastBlock = storage.getNumber(lastBlockKey) || data.seqno;
+
+            storage.set(lastBlockKey, data.seqno);
+
+            const accsToSync = [
```

Comment:
I do not like this. Let's rollback and test solution with single invalidate call.

Reply:
added new version as we discussed ✅